src/core/config/mac.h

```diff
@@ -499,4 +499,15 @@
 #define OPENTHREAD_CONFIG_MAC_SCAN_DURATION 300
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE
+ *
+ * This setting configures if the beacon payload parsing needs to be enabled in MAC. This is optional and is disabled by
+ * default because Thread 1.2.1 has removed support for beacon payloads.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE
+#define OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE 0
```

Comment:
Can we have this covered under some CI builds (basically have a CI build which enables this config).
The goal is have it covered to ensure it does not get broken by any future changes?

Maybe in `check-simulation-build-autotools` script?